app/plugin_api.go

```diff
@@ -70,6 +71,19 @@ func (api *PluginAPI) UnregisterCommand(teamId, trigger string) error {
 	return nil
 }
 
+func (api *PluginAPI) ExecuteSlashCommand(commandArgs *model.CommandArgs) (*model.CommandResponse, error) {
```

Comment:
We are using `*model.AppError` everywhere in the plugin API. I don't think it's worth to introduce an inconsistency now. (I still dream of a world without  `*model.AppError`)

Reply:
I don't really buy the consistency argument here. There are other APIs already that return just error and it really doesn't give us anything in this case.

Reply:
The only API endpoint that returns a plain error is `GetBundlePath()`. All the other ones return `*model.AppError`. Given that a mix in error types is more confusing the just using the same type one all over the place, I'm 3/5 on using `*model.AppError` here.

Reply:
No it's not. `RegisterCommand` and `UnregisterCommand` the other command APIs return `error`. I am being consistent with those APIs.

Reply:
Good point, I forgot about `RegisterCommand' and `UnregisterCommand`. Still, these are only two API methods among many.

Actually, returning `*model.AppError` does provide value here. The message shown in the webapp is the one under `Id` while the logged error is `Error()`. A plugin might want to know which message was shown in the webapp and hence needs `Id`.

Reply:
Given we resolve all inconsistencies with the plugin-api wrapper, I'm not sure the return signature matters too much. I'd personally lean towards returning `error` always, since there's no reason to propagate an earlier mistake and we're already inconsistent.

But one comment below on the return value.